index.js

```diff
@@ -191,48 +191,38 @@ NYC.prototype.addAllFiles = function () {
 }
 
 NYC.prototype.instrumentAllFiles = function (input, output, cb) {
-  var _this = this
-  var inputDir = '.' + path.sep
-  var visitor = function (filename) {
-    var ext
-    var transform
-    var inFile = path.resolve(inputDir, filename)
-    var code = fs.readFileSync(inFile, 'utf-8')
-
-    for (ext in _this.transforms) {
-      if (filename.toLowerCase().substr(-ext.length) === ext) {
-        transform = _this.transforms[ext]
-        break
-      }
-    }
+  const visitor = (filename, singleFile = false) => {
+    const inFile = path.resolve(input, filename)
+    const inCode = fs.readFileSync(inFile, 'utf-8')
 
-    if (transform) {
-      code = transform(code, { filename: inFile, relFile: inFile })
-    }
+    const extname = path.extname(filename).toLowerCase()
+    const transform = this.transforms[extname] || (code => code)
+    const outCode = transform(inCode, { filename: inFile })
 
-    if (!output) {
-      console.log(code)
-    } else {
-      var outFile = path.resolve(output, filename)
+    if (output) {
+      const subPath = singleFile || path.relative(input, path.resolve(input, filename))
```

Comment:
Could you explain what is going on with this change?

Reply:
I'm assuming that if an output dir is set, and we instrument a single file, then the new file should be written to `output/path.basename(inputFilename)`.  If singleFile is set to false then we're instrumenting a set of files, so the `path.relative(input, path.resolve(input, filename)` works to maintain the existing directory structure in the resulting instrumented directory.  It aims to replace the `var inputDir` variable that's currently reassigned in the case of instrumenting a directory.

So all that being said I'm not really happy with this solution, so I can have a go at reworking it and would be open to suggestions on fixing it.

Reply:
I'll do a little more digging on this, but I'm currently thinking there is no behaviour change here.

Reply:
Ah, ok that is breaking, it does preserve full path there.  Not sure what's desired in the long run but I'll get it back to behaving as is for now.

Reply:
Does this need a test?

Reply:
Please drop the subPath calculation for now (as well as the singleFile argument).  I would like to make sure we have tests dealing with this but I think logic change should not be part of this ES2015 refactor.  For now lets just do:
```js
const outFile = path.resolve(output, filename)
```

Reply:
So I've taken that back to be much closer to the original implementation